### PR TITLE
CLI: Manta Adding Docs and Shell complete

### PR DESCRIPTION
**Pull request by stack72**
Continuation of the Manta CLI PR

### CODE REVIEW

File: cmd/manta/cmd/docs/man/public.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+				key          = config.KeyDocManDir
+				longName     = "man-dir"
+				shortName    = "m"
+				description  = "Specify the MANDIR to use"
+				defaultValue = config.DefaultManDir
+			)
+
+			flags := parent.Cobra.PersistentFlags()
+			flags.StringP(longName, shortName, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longName))
```
Comment (sean-) [inline cmd/manta/cmd/docs/man/public.go]: Error return value is not checked

File: cmd/manta/cmd/docs/man/public.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+			viper.BindEnv(key, "MANDIR")
```
Comment (sean-) [inline cmd/manta/cmd/docs/man/public.go]: Error return value is not checked

File: cmd/manta/cmd/docs/md/public.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+				url := prefix + path.Join("/", strings.ToLower(base), "/")
+				return fmt.Sprintf(template, now, strings.Replace(base, "_", " ", -1), base, url)
+			}
+
+			linkHandler := func(name string) string {
+				base := strings.TrimSuffix(name, path.Ext(name))
+				return prefix + path.Join("/", strings.ToLower(base), "/")
+			}
+
+			doc.GenMarkdownTreeCustom(cmd.Root(), mdDir, prepender, linkHandler)
```
Comment (sean-) [inline cmd/manta/cmd/docs/md/public.go]: Error return value is not checked

File: cmd/manta/cmd/docs/md/public.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+				key          = config.KeyDocMarkdownDir
+				longName     = "dir"
+				shortName    = "d"
+				description  = "Specify the directory for generated Markdown files"
+				defaultValue = config.DefaultMarkdownDir
+			)
+
+			flags := parent.Cobra.Flags()
+			flags.StringP(longName, shortName, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longName))
```
Comment (sean-) [inline cmd/manta/cmd/docs/md/public.go]: Error return value is not checked

File: cmd/manta/cmd/docs/md/public.go [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+				key          = config.KeyDocMarkdownURLPrefix
+				longName     = "url-prefix"
+				shortName    = "p"
+				description  = "Specify the prefix for links generated by Markdown"
+				defaultValue = config.DefaultMarkdownURLPrefix
+			)
+
+			flags := parent.Cobra.Flags()
+			flags.String(longName, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longName))
```
Comment (sean-) [inline cmd/manta/cmd/docs/md/public.go]: Error return value is not checked

File: cmd/manta/cmd/docs/public.go [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+	},
+
+	Setup: func(parent *command.Command) error {
+		cmds := []*command.Command{
+			man.Cmd,
+			md.Cmd,
+		}
+
+		for _, cmd := range cmds {
+			cmd.Setup(cmd)
```
Comment (sean-) [inline cmd/manta/cmd/docs/public.go]: Error return value is not checked

File: cmd/manta/cmd/list/public.go [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+			if err != nil {
+				return err
+			}
+
+			directoryOutput, err := s.GetDirectoryListing(args)
+			if err != nil {
+				return err
+			}
+
+			cons.Write([]byte(fmt.Sprintf("Found %d directory entries", directoryOutput.ResultSetSize)))
```
Comment (sean-) [inline cmd/manta/cmd/list/public.go]: Error return value is not checked

File: cmd/manta/cmd/list/public.go [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+			for _, entry := range directoryOutput.Entries {
+				cons.Write([]byte(fmt.Sprintf("\n%s/", entry.Name)))
```
Comment (sean-) [inline cmd/manta/cmd/list/public.go]: Error return value is not checked

File: cmd/manta/cmd/root.go [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+				description = "Use a pager to read the output (defaults to $PAGER, less(1), or more(1))"
+			)
+			var defaultValue bool
+			if isatty.IsTerminal(os.Stdout.Fd()) || isatty.IsCygwinTerminal(os.Stdout.Fd()) {
+				defaultValue = true
+			}
+
+			flags := parent.Cobra.PersistentFlags()
+			flags.BoolP(longName, shortName, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longName))
```
Comment (sean-) [inline cmd/manta/cmd/root.go]: Error return value is not checked

File: cmd/manta/cmd/root.go [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+				key          = config.KeyLogLevel
+				longOpt      = "log-level"
+				shortOpt     = "l"
+				defaultValue = "INFO"
+				description  = "Change the log level being sent to stdout"
+			)
+
+			flags := parent.Cobra.PersistentFlags()
+			flags.StringP(longOpt, shortOpt, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longOpt))
```
Comment (sean-) [inline cmd/manta/cmd/root.go]: Error return value is not checked

File: cmd/manta/cmd/root.go [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+				key         = config.KeyLogFormat
+				longOpt     = "log-format"
+				shortOpt    = "F"
+				description = `Specify the log format ("auto", "zerolog", or "human")`
+			)
+			defaultValue := logger.FormatAuto.String()
+
+			flags := parent.Cobra.PersistentFlags()
+			flags.StringP(longOpt, shortOpt, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longOpt))
```
Comment (sean-) [inline cmd/manta/cmd/root.go]: Error return value is not checked

File: cmd/manta/cmd/root.go [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+				description = "Use ASCII colors"
+			)
+			defaultValue := false
+			if isatty.IsTerminal(os.Stdout.Fd()) || isatty.IsCygwinTerminal(os.Stdout.Fd()) {
+				defaultValue = true
+			}
+
+			flags := parent.Cobra.PersistentFlags()
+			flags.BoolP(longOpt, shortOpt, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longOpt))
```
Comment (sean-) [inline cmd/manta/cmd/root.go]: Error return value is not checked

File: cmd/manta/cmd/root.go [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+				key          = config.KeyUseUTC
+				longName     = "utc"
+				shortName    = "Z"
+				defaultValue = false
+				description  = "Display times in UTC"
+			)
+
+			flags := parent.Cobra.PersistentFlags()
+			flags.BoolP(longName, shortName, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longName))
```
Comment (sean-) [inline cmd/manta/cmd/root.go]: Error return value is not checked

File: cmd/manta/cmd/root.go [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+				key          = config.KeyMantaAccount
+				longName     = "account"
+				shortName    = "A"
+				defaultValue = ""
+				description  = "Account (login name). If not specified, the environment variable MANTA_USER will be used"
+			)
+
+			flags := parent.Cobra.PersistentFlags()
+			flags.StringP(longName, shortName, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longName))
```
Comment (sean-) [inline cmd/manta/cmd/root.go]: Error return value is not checked

File: cmd/manta/cmd/root.go [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+				key          = config.KeyMantaURL
+				longName     = "url"
+				shortName    = "U"
+				defaultValue = ""
+				description  = "Manta URL. If not specified, the environment variable MANTA_URL will be used"
+			)
+
+			flags := parent.Cobra.PersistentFlags()
+			flags.StringP(longName, shortName, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longName))
```
Comment (sean-) [inline cmd/manta/cmd/root.go]: Error return value is not checked

File: cmd/manta/cmd/root.go [hunk 16]
```diff
@@ -0,0 +1,205 @@
...
+				key          = config.KeyTritonSSHKeyID
+				longName     = "key-id"
+				shortName    = "K"
+				defaultValue = ""
+				description  = "This is the fingerprint of the public key matching the key specified in key_path. It can be obtained via the command ssh-keygen -l -E md5 -f /path/to/key. It can be provided via the SDC_KEY_ID or TRITON_KEY_ID environment variables."
+			)
+
+			flags := parent.Cobra.PersistentFlags()
+			flags.StringP(longName, shortName, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longName))
```
Comment (sean-) [inline cmd/manta/cmd/root.go]: Error return value is not checked

File: cmd/manta/cmd/root.go [hunk 17]
```diff
@@ -0,0 +1,205 @@
...
+			const (
+				key          = config.KeyTritonSSHKeyMaterial
+				longName     = "key-material"
+				defaultValue = ""
+				description  = "This is the private key of an SSH key associated with the Triton account to be used. If this is not set, the private key corresponding to the fingerprint in key_id must be available via an SSH Agent. It can be provided via the SDC_KEY_MATERIAL or TRITON_KEY_MATERIAL environment variables."
+			)
+
+			flags := parent.Cobra.PersistentFlags()
+			flags.String(longName, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longName))
```
Comment (sean-) [inline cmd/manta/cmd/root.go]: Error return value is not checked

File: cmd/manta/cmd/root.go [hunk 18]
```diff
@@ -0,0 +1,205 @@
...
+		}
+
+		return nil
+	},
+}
+
+func Execute() error {
+
+	rootCmd.Setup(rootCmd)
```
Comment (sean-) [inline cmd/manta/cmd/root.go]: Error return value is not checked

File: cmd/manta/cmd/root.go [hunk 19]
```diff
@@ -0,0 +1,205 @@
...
+
+	conswriter.UsePager(viper.GetBool(config.KeyUsePager))
```
Comment (sean-) [inline cmd/manta/cmd/root.go]: Error return value is not checked

File: cmd/manta/cmd/root.go [hunk 20]
```diff
@@ -0,0 +1,205 @@
...
+
+	if err := logger.Setup(); err != nil {
+		return err
+	}
+
+	for _, cmd := range subCommands {
+		rootCmd.Cobra.AddCommand(cmd.Cobra)
+		cmd.Setup(cmd)
```
Comment (sean-) [inline cmd/manta/cmd/root.go]: Error return value is not checked

File: cmd/manta/cmd/shell/autocomplete/bash/public.go [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+			const (
+				key          = config.KeyBashAutoCompletionTarget
+				longName     = "bash-autocomplete-dir"
+				defaultValue = "/etc/bash_completion.d"
+				description  = "autocompletion directory"
+			)
+
+			flags := parent.Cobra.PersistentFlags()
+			flags.String(longName, defaultValue, description)
+			viper.BindPFlag(key, flags.Lookup(longName))
```
Comment (sean-) [inline cmd/manta/cmd/shell/autocomplete/bash/public.go]: Error return value is not checked

File: cmd/manta/cmd/shell/autocomplete/public.go [hunk 22]
```diff
@@ -0,0 +1,35 @@
...
+		Short: "Autocompletion generation",
+	},
+
+	Setup: func(parent *command.Command) error {
+		cmds := []*command.Command{
+			bash.Cmd,
+		}
+
+		for _, cmd := range cmds {
+			cmd.Setup(cmd)
```
Comment (sean-) [inline cmd/manta/cmd/shell/autocomplete/public.go]: Error return value is not checked

File: cmd/manta/cmd/shell/public.go [hunk 23]
```diff
@@ -0,0 +1,35 @@
...
+		Short: "shell commands",
+	},
+
+	Setup: func(parent *command.Command) error {
+		cmds := []*command.Command{
+			autocomplete.Cmd,
+		}
+
+		for _, cmd := range cmds {
+			cmd.Setup(cmd)
```
Comment (sean-) [inline cmd/manta/cmd/shell/public.go]: Error return value is not checked

File: cmd/manta/cmd/version/public.go [hunk 24]
```diff
@@ -0,0 +1,35 @@
...
+
+var Cmd = &command.Command{
+	Cobra: &cobra.Command{
+		Use:          "version",
+		Short:        "print Joyent manta cli version",
+		SilenceUsage: true,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cons := conswriter.GetTerminal()
+			cons.Write([]byte(fmt.Sprintf("Version: %s\n", triton.UserAgent())))
```
Comment (sean-) [inline cmd/manta/cmd/version/public.go]: Error return value is not checked

File: cmd/manta/main.go [hunk 25]
```diff
@@ -0,0 +1,29 @@
...
+
+	"github.com/joyent/triton-go/cmd/manta/cmd"
+	"github.com/rs/zerolog/log"
+	"github.com/sean-/conswriter"
+)
+
+func main() {
+	defer func() {
+		p := conswriter.GetTerminal()
+		p.Wait()
```
Comment (sean-) [inline cmd/manta/main.go]: Error return value is not checked

**Comment by stack72** [comment]
This is a continuation of #142

**Pull request merged**